### PR TITLE
Addon cert manager

### PR DESCRIPTION
**Pull request by lilvinz**
* adds cert-manager as addon which should be able to provide letsencrypt certificates for ingress
* fixes a typo in addon_ingress

### CODE REVIEW

**Comment by xetys** [comment]
thanks! good job

File: cmd/addon_cert-manager.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+func NewCertmanagerAddon(cluster Cluster) ClusterAddon {
+	masterNode, err := cluster.GetMasterNode()
+	FatalOnError(err)
+	return CertmanagerAddon{masterNode: masterNode}
+}
+
+func (addon CertmanagerAddon) Install(args ... string) {
+	node := *addon.masterNode
+	_, err := runCmd(node, "helm install --name cert-manager --namespace kube-system --set ingressShim.extraArgs='{--default-issuer-name=letsencrypt-prod,--default-issuer-kind=ClusterIssuer}' stable/cert-manager")
```
Comment (pierreozoux) [inline cmd/addon_cert-manager.go]: `extraArgs='{--default-issuer-name=letsencrypt-prod,--default-issuer-kind=ClusterIssuer}'`

If you read carefully the doc, it says that this argument is needed if you want backward compatibility with `kube-lego`. (i didn't read the code to be honest)

I did test without, and it works!
https://www.youtube.com/watch?v=BXHyvPdhEFQ
(Look in the description, there are all the params used.)

**Comment by pierreozoux** [comment]
Also quick question @lilvinz @xetys what do you think about:
 - adding a default ClusterIssuer?
 - passing the email as a param to the command?

Here is a default ClusterIssuer that works:
```
apiVersion: certmanager.k8s.io/v1alpha1
kind: ClusterIssuer
metadata:
  name: letsencrypt
spec:
  acme:
    server: https://acme-v01.api.letsencrypt.org/directory
    email: certificates@example.com
    privateKeySecretRef:
      name: letsencrypt-cluster-key-pair
    http01: {}
```

Then in the ingress, we need to put these annotations:
``` 
annotations: certmanager.k8s.io/cluster-issuer: letsencrypt
```

**Comment by pierreozoux** [comment]
@lilvinz created this PR: https://github.com/xetys/hetzner-kube/pull/45/files

**Comment by lilvinz** [comment]
@pierreozoux I am using exactly that kind of ClusterIssuer and it works for me as well. Shall we add this to the README.md or change something?

File: cmd/addon_cert-manager.go (hunk 1 above)
Comment (lilvinz) [reply cmd/addon_cert-manager.go]: I think these are sane defaults. The user is always free to override them. When adding a ClusterIssues like you posted below and name it accordingly "letsencrypt-prod" it works just fine.

**Comment by pierreozoux** [comment]
@lilvinz I believe this should live upstream :)
https://github.com/kubernetes/charts/pull/3975

**Pull request merged**